coupon_incompatibility/models/coupon_program.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2021 Tecnativa - David Vidal
+# Copyright 2023 Tecnativa - Stefan Ungureanu
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class CouponProgram(models.Model):
+    _inherit = "coupon.program"
+
+    incompatible_promotion_ids = fields.Many2many(
+        comodel_name="coupon.program",
+        relation="sale_coupon_program_incompatibility_rel",
+        column1="program_id",
+        column2="incompatible_program_id",
+        inverse="_inverse_incompatible_promotion_ids",
+        string="Incompatible Promotions",
+    )
+
+    def _inverse_incompatible_promotion_ids(self):
+        """To be overriden"""
+
+    def _check_promo_code(self, order, coupon_code):
+        """To be overriden"""
```

Comment:
This method is defined in `sale_coupon`. Don't redifine it here or you'll end up with the random issues you're experimenting on the tests.
```suggestion
```